logic/pulsed/sequence_generator_logic.py

```diff
@@ -117,10 +118,20 @@ def __init__(self, config, **kwargs):
         # directory for additional generate methods to import
         # (other than logic.predefined_generate_methods)
         if 'additional_methods_dir' in config.keys():
-            if not os.path.exists(config['additional_methods_dir']):
-                self.log.error('Specified path "{0}" for import of additional generate methods '
-                               'does not exist.'.format(config['additional_methods_dir']))
-                self.additional_methods_dir = None
+            if isinstance(config['additional_methods_dir'], str):
```

Comment:
I think this should probably move to `on_activate`. If you do so, please also check for the default value `None`.

And by the way, I just realized this is an old bug.
`'additional_methods_dir'` is never in `config.keys()` because the name given to the `ConfigOption` is actually `'additional_predefined_methods_path'`

Reply:
Good point, I completely overlooked that as I only tested with the ConfigOption.
Also followed your suggestion and moved the stuff to the on_activation and cleaned up a bit more.